### DISCUSSION TITLE
Future of JavaPoet

### DISCUSSION THREAD

**Discussion by fcurts** [square/javapoet#866] [General]
JavaPoet is currently inactive with major pending PRs such as #840. @swankjesse has [stated](https://github.com/square/javapoet/pull/840#issuecomment-1004787263) that "the plan is to hand-off JavaPoet to folks that can give it the attention it deserves".

All I'm asking for is transparency around the progress on handing off JavaPoet. This will help us and others make an informed decision between the following options:

1. Wait until JavaPoet is handed off and revived.
2. Migrate off JavaPoet.

**Comment by JakeWharton** [comment]
What about?

3. Copy the sources into your codebase and maintain as part of your project.

The latest version of JavaPoet started as implementation detail of google/dagger. The original version of JavaWriter started as implementation detail of square/dagger. Having the project as implementation detail removes a huge maintenance burden as you are not subject to maintaining strict API and ABI compatibility. It's an extremely low-cost option. And it does not preclude either of your options in the future.

**Reply by fcurts** [reply]
(3) may well be an option for some users. However, we use JavaPoet in projects slated to be open-sourced later this year, and open-sourcing our own fork of JavaPoet is not an option we're considering.

**Reply by JakeWharton** [reply]
Yeah sorry I didn't mean for this to be a fork that you have to publicize. I mean literally copy the sources into your project as implementation detail (i.e., not public API). As long as your license is compatible with Apache 2.0 you can simply embed the sources into your codebase with whatever changes you need. This is how those two libraries mentioned built JavaWriter and then JavaPoet.

Even if only a temporary measure, it avoids having to _immediately_ migrate away in order to make your project available.

**Comment by swankjesse** [comment]
I’m interested in handing this project off to a new steward. Ideally it’s a team that itself uses JavaPoet, like Spring or Dagger.

**Comment by liach** [comment]
For now, there are forks with record and sealed class support at https://github.com/FabricMC/javapoet published on https://maven.fabricmc.net/net/fabricmc/javapoet/ in case you want it as a bridgegap measure

**Reply by fcurts** [reply]
That's good to know. In our case, we can only use code published to Maven Central.

**Comment by liujianpc** [comment]
您好，本人已收到您的邮件！谢谢！      祝您工作顺利，生活愉快！

**Comment by vkhodygo** [comment]
Any news so far?

**Comment by carterkozak** [comment]
Hi folks, we use Javapoet heavily and have been maintaining a fork here: https://github.com/palantir/javapoet with classes using a new package name (not a drop-in replacement). Jars are published to maven central, details in the readme :-)
